snapcraft.yaml

```diff
@@ -0,0 +1,22 @@
+name: syncthing
+version: master
+summary: Open Source Continuous File Synchronization
+description: |
+  Syncthing replaces proprietary sync and cloud services with something open,
+  trustworthy and decentralized. Your data is your data alone and you deserve
+  to choose where it is stored, if it is shared with some third party and how
+  it's transmitted over the Internet.
+
+grade: devel
+confinement: strict
+
+apps:
+  syncthing:
+    command: env GOMAXPROCS=1 syncthing -home=${SNAP_USER_DIR} -verbose -logfile=${SNAP_USER_DIR}/syncthing.log
```

Comment:
@calmh it won't be wiped out. Part of the upgrade process is to copy the previous revision's data into the new location. This gives the snap the opportunity to mutate the data in some way for the upgrade and, if it fails for some reason, to rollback to the previous revision (and its un-mutated data).

However, it's my understanding that using `$SNAP_USER_DATA` as the `--home` will also set that directory up as the default sync directory, which probably isn't what you want (due to the fact that the directory is copied upon upgrade). Ideally you could store the configuration within `$SNAP_USER_DATA`, but have raw unversioned data in `$SNAP_USER_COMMON`, which doesn't get copied for upgrades. See [Where can Ubuntu snaps write data?](http://askubuntu.com/a/762405/261753) for more information on these directories.